File: Dopamine.Common/Controls/PopupEx.cs
using System.Reflection;
using System.Windows;
using System.Windows.Controls.Primitives;

namespace Dopamine.Common.Controls
{
    public class PopupEx : Popup
    {
        private static FieldInfo fi = typeof(SystemParameters).GetField("_menuDropAlignment", BindingFlags.NonPublic | BindingFlags.Static);

        public void Open()
        {
            if (SystemParameters.MenuDropAlignment)
            {
                fi.SetValue(null, false);
                this.IsOpen = true;
                fi.SetValue(null, true);
            }
            else
            {
                this.IsOpen = true;
            }
        }

        public void Close()
        {
            this.IsOpen = false;
        }
    }
}

File: Dopamine.Controls/PopupEx.vb
Imports System.Windows.Controls.Primitives

Public Class PopupEx
    Inherits Popup

#Region "Variables"
    Private Shared mFi As Reflection.FieldInfo = GetType(SystemParameters).GetField("_menuDropAlignment", CType(Reflection.BindingFlags.NonPublic + Reflection.BindingFlags.Static, Reflection.BindingFlags))
#End Region

#Region "Public"
    Public Sub Open()
        If SystemParameters.MenuDropAlignment Then
            mFi.SetValue(Nothing, False)
            IsOpen = True
            mFi.SetValue(Nothing, True)
        Else
            IsOpen = True
        End If
    End Sub

    Public Sub Close()
        IsOpen = False
    End Sub
#End Region
End Class
